cardano-testnet/src/Testnet/Process/Cli.hs

```diff
@@ -41,84 +34,63 @@ import qualified Options.Applicative as Opt
 import           System.FilePath.Posix
 
 import           Testnet.Process.Run
+import           Testnet.Types hiding (testnetMagic)
 
 import           Hedgehog (MonadTest)
 import           Hedgehog.Extras (ExecConfig)
 import qualified Hedgehog.Extras.Test.Base as H
 import qualified Hedgehog.Extras.Test.File as H (writeFile)
 
-data KeyNames = KeyNames
-  { verificationKeyFile :: FilePath
-  , signingKeyFile :: FilePath
-  }
-
-type KeyGen a = (File (VKey a), File (SKey a))
-
 cliAddressKeyGen :: ()
   => (MonadTest m, MonadCatch m, MonadIO m, HasCallStack)
-  => TmpDir
-  -> KeyNames
-  -> m (KeyGen PaymentKey)
+  => KeyPair PaymentKey
+  -> m ()
 cliAddressKeyGen = GHC.withFrozenCallStack $ shelleyKeyGen "address" "key-gen"
 
 cliStakeAddressKeyGen :: ()
   => (MonadTest m, MonadCatch m, MonadIO m, HasCallStack)
-  => TmpDir
-  -> KeyNames
-  -> m (KeyGen StakeKey)
+  => KeyPair StakeKey
+  -> m ()
 cliStakeAddressKeyGen = GHC.withFrozenCallStack $ shelleyKeyGen "stake-address" "key-gen"
 
 cliNodeKeyGenVrf :: ()
   => (MonadTest m, MonadCatch m, MonadIO m, HasCallStack)
-  => TmpDir
-  -> KeyNames
-  -> m (KeyGen VrfKey)
+  => KeyPair VrfKey
+  -> m ()
 cliNodeKeyGenVrf = GHC.withFrozenCallStack $ shelleyKeyGen "node" "key-gen-VRF"
 
 cliNodeKeyGenKes :: ()
   => (MonadTest m, MonadCatch m, MonadIO m, HasCallStack)
-  => TmpDir
-  -> KeyNames
-  -> m (KeyGen KesKey)
+  => KeyPair KesKey
+  -> m ()
 cliNodeKeyGenKes = GHC.withFrozenCallStack $ shelleyKeyGen "node" "key-gen-KES"
 
 shelleyKeyGen :: ()
   => (MonadTest m, MonadCatch m, MonadIO m, HasCallStack)
-  => String
-  -> String
-  -> TmpDir
-  -> KeyNames
-  -> m (KeyGen x)
-shelleyKeyGen command subCommand tmpDir keyNames =
-  GHC.withFrozenCallStack $ do
-    let
-      vKeyPath = tmpDir </> verificationKeyFile keyNames
```

Comment:
This was very misleading. We were only calling this function with an absolute paths in keyNames which resulted in `tmpDir` being completely ignored. That's a property of `</>`:
```haskell
"some/leading/path" </> "/absolutepath.txt" == "/absolutepath.txt"
```

Reply:
Nice catch! How did you find this?

Reply:
Some time ago when I was refactoring path building in testnet, I was quite surprised with the results.